packages/devtools_app/lib/src/service_manager.dart

```diff
@@ -587,42 +587,52 @@ class ServiceExtensionManager {
         break;
       case 'Flutter.ServiceExtensionStateChanged':
         final String name = event.json['extensionData']['extension'].toString();
-        final String valueFromJson =
+        final String valueString =
             event.json['extensionData']['value'].toString();
+        await _updateServiceExtensionForStateChange(name, valueString);
+        break;
+      case 'HttpTimelineLoggingStateChange':
+        final name = extensions.httpEnableTimelineLogging.extension;
+        final valueString = event.json['extensionData']['enabled'].toString();
+        await _updateServiceExtensionForStateChange(name, valueString);
+    }
+  }
 
-        final extension = extensions.serviceExtensionsAllowlist[name];
-        if (extension != null) {
-          final dynamic value = _getExtensionValueFromJson(name, valueFromJson);
-
-          final enabled =
-              extension is extensions.ToggleableServiceExtensionDescription
-                  ? value == extension.enabledValue
-                  // For extensions that have more than two states
-                  // (enabled / disabled), we will always consider them to be
-                  // enabled with the current value.
-                  : true;
-
-          await setServiceExtensionState(
-            name,
-            enabled,
-            value,
-            callExtension: false,
-          );
-        }
+  Future<void> _updateServiceExtensionForStateChange(
+    String name,
+    String valueString,
+  ) async {
+    final extension = extensions.serviceExtensionsAllowlist[name];
+    if (extension != null) {
+      final dynamic value = _getExtensionValue(name, valueString);
+      final enabled =
+          extension is extensions.ToggleableServiceExtensionDescription
+              ? value == extension.enabledValue
+              // For extensions that have more than two states
+              // (enabled / disabled), we will always consider them to be
+              // enabled with the current value.
+              : true;
+
+      await setServiceExtensionState(
+        name,
+        enabled,
+        value,
+        callExtension: false,
+      );
     }
   }
 
-  dynamic _getExtensionValueFromJson(String name, String valueFromJson) {
+  dynamic _getExtensionValue(String name, String valueString) {
```

Comment:
maybe valueString -> encodedValue here and elsewhere if `value` is confusing.

Reply:
done